docs-src/typescript/how-to-use-worker-versioning-in-typescript.md

```diff
@@ -0,0 +1,83 @@
+---
+id: how-to-use-worker-versioning-in-typescript
+title: How to use Worker Versioning in TypeScript
+sidebar_label: Worker Versioning
+description: Version your TypeScript Workers by using build ID-based versioning
+tags:
+  - typescript
+  - how-to
+---
+
+To use Worker Versioning in TypeScript, you need to do the following:
+
+1. Determine and assign a Build ID to your built Worker code, and opt in to versioning.
+2. Tell the Task Queue your Worker is listening on about that Build ID, and whether its compatible with an existing Build ID.
+
+### Assign a Build ID to your Worker
+
+Let's say you've chosen `deadbeef` as your Build ID, which might be a short git commit hash (a reasonable choice as Build ID).
+To assign it in your Worker code, assign the following Worker Options:
+
+```typescript
+// ...
+const worker = await Worker.create({
+  taskQueue: 'your_task_queue_name',
+  buildId: '1.0',
+  useVersioning: true,
+  // ...
+});
+// ...
+```
+
+That's all you need to do in your Worker code.
+Importantly, if you start this Worker, it won't receive any tasks.
+That's because you need to tell the Task Queue about your Worker's Build ID first.
+
+### Tell the Task Queue about your Worker's Build ID
+
+Now you can use the SDK (or the Temporal CLI) to tell the Task Queue about your Worker's Build ID.
```

Comment:
I think I'm more likely to use CLI in CI, could you show that example as well?

Reply:
Problem is I don't want to repeat that in each language guide. Given this contextually is about using the language SDK parts, I want to stick to that.

Reply:
Could you link to a new section in the CLI docs from each lang guide that has the equivalent examples & explanatory text? 

https://docs.temporal.io/cli/task-queue/

Reply:
Sure, but are you fine with me doing that in a subsequent PR?

Reply:
sure!